Patches Master Build & Other Additions

GlitchOo opened the pull request:
**Pull Request Description**

-Fixes current master build with successful pnpm install && pnpm build
-Fixes a Native Execution error with the phone model and textures (kvpservice)
-Includes PR https://github.com/project-error/npwd/pull/997
-Adjusts the the position of the popper modal for images on the twitter app (centers on screen)
-Added image upload support for "upload.io"

**Pull Request Checklist**:
* [✓] Have you followed the guidelines in our Contributing document and Code of Conduct?
* [✓] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/project-error/npwd/pulls) for the same update/change?
* [ ✓] Have you built and tested NPWD in-game after the relevant change?

I've been testing it for over 48 hours now and everything is working perfectly fine

AvarianKnight commented:
Going to wait to review this until after Mycrofts pr gets reviewed

GlitchOo commented:
Added the ability to post twitter posts to discord via webhook with the hook set in a convar
Ex:
set TWITTER_WEBHOOK "URL"

Also added bool setting to twitter config that allows just an image to be posted to twitter without typing a message. (Disabled by default)

Opy12345 commented:
how can i download the phone?

GlitchOo commented:
> how can i download the phone?

No official release of these changes yet as it's still pending a merge into the master branch. However I have made the last build of my changes available here: https://github.com/GlitchOo/npwd/releases/tag/1.7.2-G if you want them :)

Opy12345 commented:
> > how can i download the phone?
> 
> No official release of these changes yet as it's still pending a merge into the master branch. However I have made the last build of my changes available here: https://github.com/GlitchOo/npwd/releases/tag/1.7.2-G if you want them :)

ok so just install the zip file there?

GlitchOo commented:
This latest push of changes does the following. All have been tested and are working.
- Enabled Dark chat (Fixed a crashing issue caused by missing svg icon and 
       patched failed to fetch members in group chat when offline
- Cleaned up contact lists (removed any code that wasn't needed or being used
- Made the call and text buttons functional on the contact view page
      https://gyazo.com/c1afa1c26ea17c5f1e267dc8688e1244
- Patched routing on the dialer app making contacts non responsive
      https://gyazo.com/c34a6ce352a7872144642de88a21a52e
- Fixed the position where the phone spawns in the players hands causing it to clip in hand
      https://i.gyazo.com/e19e4ad1b41b54b67f9e83f6537adfdc.png
- That's all I can think of that I've changed in the last commits since the last successful workflow tests

This should bring everything up to a fully functional phone and build again

GlitchOo commented:
No idea when this will be approved and merged for the meantime you can get the unofficial build release here:
https://github.com/GlitchOo/npwd/releases/tag/1.7.3-G

Opy12345 commented:
> No idea when this will be approved and merged for the meantime you can get the unofficial build release here: https://github.com/GlitchOo/npwd/releases/tag/1.7.3-G

ok thank you.

GlitchOo commented:
Okay this should be the last of the phones current functions now working.
https://gyazo.com/d946b201f8231e830688858a5088a769

GlitchOo commented:
Added infinite scrolling to the twitter app
https://gyazo.com/63c2b2540fdd9729df14c92cf9bcd869
I guess all that's left is a merge into the master 👀👀👀

ArielZ123 commented:
> Added infinite scrolling to the twitter app https://gyazo.com/63c2b2540fdd9729df14c92cf9bcd869 I guess all that's left is a merge into the master 👀👀👀

Amazing let us know here when we can download it.

AvarianKnight commented on apps/game/server/twitter/twitter.service.ts:
```diff
@@ -6,12 +6,15 @@ import { reportTweetToDiscord } from '../misc/discord';
 
 class _TwitterService {
   private readonly twitterDB: _TwitterDB;
+  private readonly DISCORDWH: string;
```
Can you change this to `DISCORD_WEBHOOK` the abbreviation might confuse some

GlitchOo replied:
changed

AvarianKnight commented on apps/phone/src/apps/contacts/components/views/ContactInfo.tsx:
```diff
@@ -81,10 +159,8 @@ const ContactsInfoPage: React.FC = () => {
-        onClick={() => history.goBack()}
-        className="mt-4 ml-4 rounded-md px-3 py-1 hover:bg-gray-200 hover:dark:bg-neutral-800"
-      >
+      <SendMoneyModal open={contactPayModal} closeModal={() => setcontactPayModal(false)} openContact={number} />
```
This should probably be added as a config for the people who enable this integration

GlitchOo replied:
it already is

AvarianKnight commented on apps/phone/src/apps/contacts/hooks/useModal.tsx:
```diff
@@ -0,0 +1,8 @@
+import { showMoneyModal } from './state';
+
+export const useModal = () => {
+  const [contactPayModal, setcontactPayModal] = useRecoilState<boolean>(showMoneyModal);
```
nit: this should be setContactPayModal to keep consistent with the rest of the project

GlitchOo replied:
changed

AvarianKnight commented on apps/phone/src/apps/contacts/hooks/state.ts:
```diff
@@ -55,4 +61,4 @@ export const useContactsValue = () => useRecoilValue(contacts);
 
 export const useContactFilterInput = () => useRecoilState(filterInput);
-export const useSetContactFilterInput = () => useSetRecoilState(filterInput);
+export const useSetContactFilterInput = () => useSetRecoilState(filterInput)
```
Why remove the semicolon ☹️

GlitchOo replied:
🤐

AvarianKnight commented on apps/phone/src/apps/twitter/components/AddTweetModal.tsx:
```diff
@@ -80,8 +80,15 @@ const AddTweetModal = () => {
+      }
+    }
+
+    if (cleanedMessage.length > 0)
```
Why did you remove the return here?

GlitchOo replied:
I didn't, I modified it to to check to see if they have enabled the new setting in the twitter section of the config (disabled by default) to check if they wish to let users post only an image with no message. Otherwise it continues as normal. Not sure if I explained that correctly but hopefully you get the point

AvarianKnight replied:
shouldn't this just be `if (cleanedMessage.length > 0 && !isValidMessage(cleanedMessage)) return;` then?

The way this is formatted kinda just confuses me

GlitchOo replied:
cleaned up

AvarianKnight commented on apps/phone/src/apps/twitter/components/tweet/TweetList.tsx:
```diff
@@ -2,20 +2,47 @@ import { memo, useState } from 'react';
+  const [page, setPage] = useState(0)
+  const [hasMore, setHasMore] = useState((tweetsData.length >= 20 ? true : false))
+
+  const loadmore = () => {
```
This doesn't solve the original issue the phone had with InfiniteScroll before, reference: https://github.com/project-error/npwd/issues/639#issuecomment-1050162876

Any infinite scrolling that would be added would need to be able to cull elements which no major library currently offers.

GlitchOo replied:
I see, I'm pretty sure this is easily solvable thinking about it. ill remove it for the meantime and set it back to the hardcoded limit of 25 and revisit this later. I have a pretty good idea how I could easily get around this problem.

AvarianKnight commented on apps/game/server/contacts/contacts.controller.ts:
```diff
@@ -1,7 +1,29 @@
+const exps = global.exports;
+
+onNetPromise<ContactPay, Contact>(ContactEvents.PAY_CONTACT, (reqObj, resp) => {
+  PlayerService.getIdentifierByPhoneNumber(reqObj.data.number, true).then((response) => {
```
This should be behind an optional config

GlitchOo replied:
again this already is from the client side

AvarianKnight commented on apps/phone/src/apps/contacts/components/views/ContactInfo.tsx:
```diff
@@ -73,6 +141,16 @@ const ContactsInfoPage: React.FC = () => {
 
+  const openpayModal = () => {
+    if (ResourceConfig.general && ResourceConfig.general.useResourceIntegration && 
+        ResourceConfig.contacts && ResourceConfig.contacts.frameworkPay) {
```
Couldn't you use option chaining here? Feel free to correct me if I'm wrong
`ResourceConfig?.general?.useResourceIntegration`

GlitchOo replied:
You are correct, I meant to do this before pushing it but I guess I just moved on the next "to do"

GlitchOo commented on .npmrc:
```diff
@@ -0,0 +1 @@
+public-hoist-pattern[]=*types*
```
yeah I dont know to be honest, I think it happened when the npm for lodash was broken and I had to load it from github directly to get it to work for the build... it's removed though lol

AvarianKnight commented on apps/phone/src/index.tsx:
```diff
@@ -7,6 +7,7 @@ import dayjs from 'dayjs';
 import relativeTime from 'dayjs/plugin/relativeTime';
 import updateLocale from 'dayjs/plugin/updateLocale';
 import { Theme as MaterialUITheme } from '@mui/material';
+import { RewriteFrames } from '@sentry/integrations';
```
Unused import

GlitchOo replied:
I never added this.. this was from the PR before mine. Ill remove it though as it shouldn't be there

GlitchOo replied:
removed

AvarianKnight commented on apps/phone/src/os/apps/hooks/useApps.tsx:
```diff
@@ -8,6 +8,7 @@ import { useSettingsValue } from '../../../apps/settings/hooks/useSettings';
 import { IconSetObject } from '@typings/settings';
 import { useRecoilValue } from 'recoil';
 import { phoneState } from '@os/phone/hooks/state';
+import { extname } from 'path';
```
Unused import

GlitchOo replied:
Again same... never added this but ill remove

GlitchOo replied:
removed

AvarianKnight commented on packages/database/src/darkchat/index.ts:
```diff
@@ -111,11 +111,7 @@ export class _DarkchatDB {
-    userIdentifier: string,
-    isOwner: boolean,
-  ): Promise<number> {
+  async joinChannel(channelIdentifier: string, userIdentifier: string, isOwner: boolean): Promise<number> {
```
This shouldn't be collapsed, eslint show throw warnings/errors about this

GlitchOo replied:
also no warnings or errors here as far as im aware. this is the first im ever hearing about this? was this a previous version because I upgraded eslint in the packages at some point im pretty sure

GlitchOo replied:
looking at much of the db class exports theyre all written in a similar fashion. is such a change necessary? if thats the case then shouldn't all of the classes be modified as well?

AvarianKnight commented on typings/config.ts:
```diff
@@ -1,3 +1,5 @@
+import { Config } from "prettier";
```
Unused import

GlitchOo replied:
> You also need to resolve the merge conficts

i believe i can do that, however I am pretty sure they were caused by the PR before this

GlitchOo replied:
removed

AvarianKnight commented on apps/game/client/functions.ts:
```diff
@@ -33,27 +33,17 @@ export const newPhoneProp = async () => {
     const [x, y, z] = GetEntityCoords(playerPed, true);
     global.phoneProp = CreateObject(GetHashKey(phoneModel), x, y, z + 0.2, true, true, true);
     const boneIndex = GetPedBoneIndex(playerPed, 28422);
-    AttachEntityToEntity(
```
This shouldn't be collapsed, eslint should warn/error about this

GlitchOo replied:
no warning or errors unless im just missing something

GlitchOo replied:
i made the change anyways, not sure what difference it actually makes

AvarianKnight reviewed (changes requested):
Some things I missed in my initial review

AvarianKnight commented:
You also need to resolve the merge conficts

GlitchOo commented:
Okay I've resolved the conflicts... 188 conflicts in a pnpm lock file is on the list of one of the not so amazing things I've had to do lol

itschip commented on apps/phone/src/apps/twitter/components/AddTweetModal.tsx:
```diff
@@ -80,8 +80,14 @@ const AddTweetModal = () => {
     const cleanedMessage = clean(message.trim());
-    if (cleanedMessage.length === 0) return;
-    if (!isValidMessage(cleanedMessage)) return;
+    if (cleanedMessage.length === 0) {
```
You can just check if `cleanedMessage` is false instead of the length :)

GlitchOo replied:
yeah I suppose i just left what was there from whoever wrote the code. ill change it now and push

GlitchOo replied:
cleaned up, should be better

itschip commented on package.json:
```diff
@@ -40,5 +40,8 @@
     }
+  },
+  "dependencies": {
+    "lodash": "https://github.com/lodash/lodash.git#npm"
```
This shouldn't be here. Uninstall it, and install it in the package where it belong :)

GlitchOo replied:
yeah think this happened when i was having issues with getting the correct lodash ill clean that up now

GlitchOo replied:
fixed

itschip commented on package.json:
```diff
@@ -16,7 +16,8 @@
     "rollup-plugin-esbuild": "^4.10.2",
-    "turbo": "^1.8.3"
+    "turbo": "^1.8.3",
+    "lodash": "https://github.com/lodash/lodash.git#npm"
```
What are you using lodash for? I don't see any usage with it. 

and if it's only used in `apps/phone`, it should be installed in that package.json

GlitchOo replied:
It's a dependency for eslint and a few other packages in the dev dependency. The dependency downloaded by npn is broken and doesn't have the modules required so the build will just fail

AvarianKnight reviewed (approved):
LGTM

TasoOneAsia commented:
Taking a look at this a little bit after the fact and want to reiterate good work :) Not often do I see such a high-quality PR

TonybynMp4 commented:
damn the messiah complimented you that's.. something 😲

GlitchOo commented:
> Taking a look at this a little bit after the fact and want to reiterate good work :) Not often do I see such a high-quality PR

I appreciate that, thank you
